src/request/component-models/LoggingComponentModel.ts

```diff
@@ -44,6 +71,28 @@ export class LogMessageModel implements ILogMessageModel {
         return this._spans;
     }
 
+    private static indexOf(value: string, term: string, window: number) {
+        for (let i = 0; i < value.length && i < window; i++) {
+            if (value[i] === term) {
+                return i;
+            }
+        }
+        
+        return -1;
+    }
+
+    private static lastIndexOf(value: string, term: string, window: number) {
+        const lastWindowIndex = value.length - window;
+        
+        for (let i = value.length - 1; i >= 0 && i >= lastWindowIndex; i--) {
+            if (value[i] === term) {
+                return i;
+            }
+        }
+        
+        return -1;
+    }
+
```

Comment:
Just wondering if we should store the open/closed state on the underlying `_message` object? If we did it means that as someone moves between tabs, that messages would still be open/closed when they move back to the logging tab.